### PR TITLE
feat(i18n - git-changelog): add locale support for contributors

### PR DESCRIPTION
**Pull request by Ampernic**
Our team took on a new project in which there was a need to localize contributor names for better internationalization.

I would like to propose my own implementation of this functionality.

In `Author` (`Contributor`) type inside `mapAuthors` option we create `i18n` field with the following structure:
```ts
i18n: {
   'locale-code': 'name',
}
```

Example:
```ts
mapAuthors: [
   {
       name: "Anton Politov",
       i18n: {
           'ru-RU': 'Антон Политов',
       },
      ...
   }
]
```

Maybe it can be helpful for someone

### CODE REVIEW

File: packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue [hunk 1]
```diff
@@ -11,6 +12,8 @@ const options = defu(inject(InjectionKey, {}), defaultOptions)
 const { t } = useI18n()
 const { authors, useHmr } = useChangelog()
 
+const { localeIndex } = useData()
```
Comment (nekomeowww) [inline packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue:15]: May I ask what is `localeIndex`? VitePress doesn't have much documentations about this field, and I not quite sure what's the difference between `lang` and `localeIndex` after looking at the source code of VitePress with the following Pull Requests:

- https://github.com/vuejs/vitepress/pull/3862/files
- https://github.com/vuejs/vitepress/pull/1339/files#diff-5a628d1968485980dfc86e06a4a9ed3ed19a912261045f65d3e343e57ad91a97R14

Comment (Ampernic) [reply packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue:15]: I'm sorry, but i cant understand this too)
This variables has the same values in live usage. 
The choice was only due to the fact that it was the one I had seen used elsewhere.

Comment (Ampernic) [reply packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue:15]: maybe lang can be more useful in i18n

`lang` use only lang codes
`localeIndex` can show "root" for any locale

Comment (nekomeowww) [reply packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue:15]: 1. Perhaps we could file an issue to VitePress to request more explanation on this field (and prevent anti-patterns and bad practices).
2. Besides all of that, VitePress doesn't have much documentations about `root` too... Currently, all the Nolebase Integrations are using lang or helper-calculated `lang` from paths and urls. Do you have any of `root` used to determine and configure for locales that I could reference to for better understanding?

Much appriciated.

cc @northword

**Comment by nekomeowww** [comment]
Just merged #283 , updating against `main` branch or rebasing with `main` is recommended. (Just recommendation, feel free to do it at any time).

**Comment by Ampernic** [comment]
> Just merged #283 , updating against `main` branch or rebasing with `main` is recommended. (Just recommendation, feel free to do it at any time).

Ok, i`ll change `localeIndex` to `lang` and rebase in several minutes.

File: packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue (hunk 1 above)
Comment (northword) [reply packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue]: I'm sorry, I really don't know much about the i18n related parts of VitePress, so I can't give much advice.

From the information mentioned so far, both `lang` and `langIndex` seem to make the feature work?

If it is not necessary to use `langIndex` to work properly, I think `lang` is a better choice to keep the Nolebase plugin options consistent.

**Review by nekomeowww — APPROVED** [review]
LGTM! Thanks for contributing, let's wait for @northword to take a look too.

**Review by northword — APPROVED** [review]

**Comment by northword** [comment]
Oh, we may need to add documentation for this option at some other time.

**Comment by nekomeowww** [comment]
Added `documentation` label to track. @Ampernic would you like to write the English version of the documentation as well?

**Pull request merged**